lib/database/model/sthDatabaseNaming.js

```diff
@@ -0,0 +1,248 @@
+      break;
+  }
+  maxIndexNamespaceSizeInBytes = MAX_INDEX_NAMESPACE_SIZE_IN_BYTES - bytesCounter.count(databaseName) -
+    bytesCounter.count('.') - bytesCounter.count(sthConfig.COLLECTION_PREFIX) - compoundIndexBytes;
+  maxNamespaceSizeInBytes = MAX_NAMESPACE_SIZE_IN_BYTES - bytesCounter.count(databaseName) - bytesCounter.count('.') -
+    bytesCounter.count(sthConfig.COLLECTION_PREFIX) - bytesCounter.count('.aggr');
+  return Math.min(maxIndexNamespaceSizeInBytes, maxNamespaceSizeInBytes);
+}
+
+/**
+ * Returns the size of the namespace (for the collection where the aggregated data is stored) in bytes for certain
+ *  database name and the collection name where the raw data is stored
+ * @param  {String} datebaseName          The database name
+ * @param  {String} rawCollectionName The name of the collection where the raw data is stored
+ * @return {Number} The size in bytes of the namespace (for the collection where the aggregated data is stored)
+ *  in bytes
+ */
+function getNamespaceSizeInBytes(datebaseName, rawCollectionName) {
+  return bytesCounter.count(datebaseName) + bytesCounter.count(sthConfig.COLLECTION_PREFIX) +
+    bytesCounter.count(rawCollectionName) + bytesCounter.count('.aggr');
+}
+
+/**
+ * Generates a hash based on an input and a maximum number of bytes
+ * @param {String} input The input to generate the hash from
+ * @param {Number} limit The maximum number of bytes of the hash
+ */
+function generateHash(input, limit) {
+  var shasum = crypto.createHash('sha512');
+  shasum.update(input);
+  var hash = shasum.digest('hex');
+  if (limit) {
+    hash = hash.substr(0, limit);
+  }
+  return hash;
+}
+
+/**
+ * Returns the name of the collection which will store the raw events
+ * @param  {Object}  params   Params object including the following properties:
+ *                              - service: The service
+ *                              - servicePath: The service path
+ *                              - entityId: The entity id
+ *                              - entityType: The entity type
+ *                              - attrName: The attribute name
+ * @returns {String}          The raw data collection name
+ */
+function getRawCollectionName(params) {
+  var collectionName4Events;
+  var databaseName = getDatabaseName(params.service),
+      servicePath = params.servicePath,
+      entityId = params.entityId,
+      entityType = params.entityType,
+      attrName = params.attrName;
+
+  switch (sthConfig.DATA_MODEL) {
+    case sthConfig.DATA_MODELS.COLLECTION_PER_SERVICE_PATH:
+      collectionName4Events = (sthConfig.NAME_ENCODING ?
+        sthDatabaseNameCodec.encodeCollectionName(servicePath) :
+        servicePath);
+      break;
+    case sthConfig.DATA_MODELS.COLLECTION_PER_ENTITY:
+      collectionName4Events =
+        (sthConfig.NAME_ENCODING ? sthDatabaseNameCodec.encodeCollectionName(servicePath) : servicePath) +
+          sthConfig.NAME_SEPARATOR +
```

Comment:
This will always use `xffff` as concatenator, even in the case `name_encoding=false`. As a STH user, I would expect the old underscore as concatenator if `name_encoding=false`.